lib/console.js

```diff
@@ -249,6 +264,108 @@ Console.prototype.groupEnd = function groupEnd() {
     this[kGroupIndent].slice(0, this[kGroupIndent].length - 2);
 };
 
+const KKEY = 'Key';
+const VKEY = 'Values';
+const IKEY = '(index)';
+const ITKEY = '(iteration index)';
+
+const inspect = (v) => {
+  const opt = { depth: 0, maxArrayLength: 3 };
+  if (v !== null && typeof v === 'object' &&
+      !ArrayIsArray(v) && ObjectKeys(v).length > 2)
+    opt.depth = -1;
+  return util.inspect(v, opt);
+};
+
+const increasingArray = (n) => ArrayFrom({ length: n }, (_, i) => inspect(i));
+
+// https://console.spec.whatwg.org/#table
+Console.prototype.table = function(tabularData, properties) {
+  if (tabularData == null ||
+      ['symbol', 'boolean', 'number', 'string'].includes(typeof tabularData) ||
+      (ArrayIsArray(tabularData) && tabularData.length > 20) ||
+      ObjectKeys(tabularData) > 20)
+    return this.log(tabularData);
+
+  const mapIter = isMapIterator(tabularData);
+  let setIter;
+  if (mapIter)
+    tabularData = previewMapIterator(tabularData);
+  else if (setIter = isSetIterator(tabularData))
+    tabularData = previewSetIterator(tabularData);
+
+  const final = (k, v) => this.log(cliTable(k, v));
+
+  if (mapIter || isMap(tabularData)) {
+    const kkeyArr = [];
+    const vkeyArr = [];
+    for (const [k, v] of tabularData) {
+      kkeyArr.push(inspect(k));
+      vkeyArr.push(inspect(v));
+    }
+    return final([
+      ITKEY, KKEY, VKEY
+    ], [
+      increasingArray(tabularData.length || tabularData.size),
+      kkeyArr,
+      vkeyArr,
+    ]);
+  } else if (setIter || isSet(tabularData)) {
```

Comment:
```js
} else
if {
```
???

Reply:
Each of the branches seems to return. So these blocks can be rewritten as:

```js
if (condition) {
  return result;
}

if (condition2) {
  return result2;
}

```

Reply:
No, there is a `return` in the if above. So the else is obsolete and it is easier to read by moving the next if a tiny bit down:

```js
if (...) {
  ...
  return ...
}

if (...) {
 ...
}
```

Reply:
On that note, we were even going to add an eslint rule for this but there was too much old code that didn't conform. It would be nice to introduce more.

Reply:
@apapirovski you beat me :D